### DISCUSSION TITLE
Market Management Framework

### DISCUSSION THREAD

**Discussion by joshhirsh** [zed-industries/zed#18881] [Feature Requests]
Hello, I was just curious if Zed will be adopting the marketplace management framework that VS Code is about to rollout?

**Comment by SomeoneToIgnore** [comment]
The search does not show anything useful, so please share more context on what this is.

**Reply by joshhirsh** [reply]
Will Zed be able to use the same extension marketplace as VS Code?

**Reply by SomeoneToIgnore** [reply]
There's https://github.com/zed-industries/zed/issues/4845 as an issue to track the related thing.

Is it possible to use "same extension marketplace as VS Code" legally, I wonder? 
I expect some legal difficulties given our different licensing policy and VSCodium's example.

But besides those, the technical reasons are making this quite unlikely to happen:
* Zed follows the webassembly way as the base for the extension support, which is not compatible with regular VSCode extensions.
Zed does not know enough about JavaScript or Typescript and has no runtime to run its plugins inside.

* Current plugins in Zed are very limited and do not allow text editing, UI elements usage or customization, etc. — and making that happen might take quite a while.
The roots are planted for some "edit a text from the extension" right now: https://github.com/zed-industries/zed/pull/17809
So it's very far from the surface a typical VSCode extension plugin requires.

I do not believe Zed ever supports any VSCode extensions out of the box, but maybe somebody's amazing project makes this happen later. Nothing official is planned this year at least.

**Reply by joshhirsh** [reply]
Sweet, thanks for your help!
